src/file-upload/file-item-edit.tsx

```diff
@@ -54,19 +56,23 @@ export const FileItemEdit = ({
         setFormattedName(getTruncatedText(name));
     }, [wrapperWidth]);
 
+    useEffect(() => {
+        setCurrentDescription(fileItem.description || "");
+    }, [fileItem]);
+
     // =========================================================================
     // EVENT HANDLERS
     // =========================================================================
     const handleSave = () => {
-        if (textareaRef.current) {
-            onEdit(textareaRef.current.value || "");
-        } else {
-            onEdit(undefined);
-        }
+        onSave(textareaRef.current.value);
+    };
+
+    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        setCurrentDescription(event.target.value.trim());
```

Comment:
trimming in onChange can result in some funny behaviour, I think the trim could be done on save + in the disabled check instead

https://github.com/LifeSG/react-design-system/assets/16853062/2647bc73-1119-4999-9229-9a6638e3b9b6

Reply:
Yea i think typing space in between text could also be an issue. Should have figured. Careless. Yes i think trimming on save will be a better way. For the disabled check, i think we can check for empty values explicitly